pandas/core/arrays/string_arrow.py

```diff
@@ -663,7 +664,10 @@ def _convert_int_dtype(self, result):
 
     def _cmp_method(self, other, op):
         result = super()._cmp_method(other, op)
```

Comment:
Shouldn't this be fixed in the `ArrowExtensionArray._cmp_method`?

Reply:
No, the arrow dtypes will return an arrow array with bool dtype that has NA values in those places, which is correct, this fix is to emulate object dtype